### PR TITLE
Fixes and Optimisations

### PR DESCRIPTION
**Pull request by lordakshaya**
1. Fixed build mechanism for hook on linux OS - the regex flag -E isnt required (and isnt supported) on linux.
2. Exposed Cordova prepare method for optimistic building

### CODE REVIEW

File: taco-team-build.js [hunk 1]
```diff
@@ -90,7 +90,9 @@ function applyExecutionBitFix(platforms) {
         var platformCordovaDir = "platforms/" + platform + "/cordova";
         
         if (utilities.fileExistsSync(platformCordovaDir)) {
-            script += "find -E " + platformCordovaDir + " -type f -regex \"[^.(LICENSE)]*\" -exec chmod +x {} +\n"
+            // Disable -E flag for non-OSX platforms
+            var regex_flag = process.platform == 'darwin' ? '-E' : '';
```
Comment (dlevine82) [inline taco-team-build.js]: The check directly above exits the function if we're not on OSX, so right now this additional check won't do anything. We would need to alter the check at the top of the function to continue for both OSX *and* Linux.

**Comment by Chuxel** [comment]
Dan is correct - Generating the ipa is recent.  There is code now that skips it when a version of cordova-ios is found that auto-generates.

**Comment by lordakshaya** [comment]
Oh ok. I will update my PR and send it over again. Thanks for the feedback.

File: taco-team-build.js (hunk 1 above)
Comment (ryuyu) [inline taco-team-build.js]: I think if this is addressed, we will be a in a good spot to merge this in.

Comment (lordakshaya) [inline taco-team-build.js]: @dlevine82 @ryuyu its done. please review.

Comment (Bostonncity) [inline taco-team-build.js]: Oke

**Pull request merged**